### PR TITLE
Add basic integration test for segment replication

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description
This PR adds an integration test to do sanity for segment replication. 
 
### Issues Resolved
https://github.com/opensearch-project/k-NN/issues/921
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java [hunk 1]
```diff
@@ -55,6 +55,30 @@ public static void setUpClass() throws IOException {
...
+     * This test performs basic validation steps such as index, search in order to verify that kNN works with Segment
+     * replication.
+     * @throws Exception
+     */
+    public void testSegmentReplication() throws Exception {
+        String field1 = "field1";
+        String field2 = "field2";
+        createKnnIndex(INDEX_NAME, getKNNSegrepIndexSettings(), createKnnIndexMapping(Arrays.asList(field1, field2), Arrays.asList(2, 2)));
+
+        Float[] vector = { 6.0f, 7.0f };
```
Comment (martin-gaievski) [inline src/test/java/org/opensearch/knn/index/OpenSearchIT.java:68]: better use different vectors for each doc, it should be fine as you're not checking scores, just overall count

**Review by martin-gaievski — COMMENTED** [review]
Is this the breaking change? Seems it doesn't work with rolling upgrades, or that is work in progress?

Are you going to add more tests? Seems this one uses basic settings: 1 shard, 1 replica. Also do we need to check scores and order of doc ids in result? I'm not sure what can be a side-effect of segment replication not working correctly.

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java [hunk 2]
```diff
@@ -55,6 +55,30 @@ public static void setUpClass() throws IOException {
...
+
+        addKnnDoc(INDEX_NAME, "2", field1, vector);
+        addKnnDoc(INDEX_NAME, "3", field1, vector);
+        addKnnDoc(INDEX_NAME, "4", field1, vector);
+
+        addKnnDoc(INDEX_NAME, "5", field2, vector);
+        addKnnDoc(INDEX_NAME, "6", field2, vector);
+
+        refreshAllIndices();
+        assertEquals(6, getDocCount(INDEX_NAME));
```
Comment (naveentatikonda) [inline src/test/java/org/opensearch/knn/index/OpenSearchIT.java:79]: Is this the only assertion to validate segment replication ?

Comment (dreamer-89) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java:79]: Yes, this is the only assertion as it ensures the background segment syncs from primary to replica is working.

**Comment by dreamer-89** [comment]
> Is this the breaking change? Seems it doesn't work with rolling upgrades, or that is work in progress?
> 
> Are you going to add more tests? Seems this one uses basic settings: 1 shard, 1 replica. Also do we need to check scores and order of doc ids in result? I'm not sure what can be a side-effect of segment replication not working correctly.

Thank you @martin-gaievski for reviewing this PR. 

This PR on adding tests is work related to [bug on core](https://github.com/opensearch-project/OpenSearch/issues/7781) which impacts OS 2.7.0 version. These tests are added to provide basic validation that kNN indices works with segment replication and will also supplement testing for changes we are doing in https://github.com/opensearch-project/OpenSearch/issues/3881

Added more details in https://github.com/opensearch-project/k-NN/pull/927

**Comment by martin-gaievski** [comment]
@dreamer-89 looks like there are no more comments for code itself, could you please re-run all the CI checks

**Comment by navneet1v** [comment]
@dreamer-89 are you still working on this PR?

**Comment by dreamer-89** [comment]
> @dreamer-89 are you still working on this PR?

Thanks @navneet1v @martin-gaievski for checking in. Please let me know if you have any open questions/concerns regarding this change. If not, lets move forward with PR approval and code merge.

**Comment by navneet1v** [comment]
@dreamer-89 the build is broken please fix the build.

**Pull request closed without merging**